src/components/content-editor/event-trigger-editor.tsx

```diff
@@ -448,6 +452,85 @@ class EventTriggerEditor extends React.Component<Props, State> {
     );
   }
 
+  /**
+   * Renders execute web script settings
+   */
+  private renderExecuteWebScriptSettings = () => {
+    const { classes } = this.props;
+    if (this.getSelectedEventActionType() !== ExhibitionPageEventActionType.ExecuteWebScript) {
+      return;
+    }
+
+    const selectedPageEvent = this.getCurrentPageEvent();
+    if (!selectedPageEvent) {
+      return null;
+    }
+
+    const webViewProperty = selectedPageEvent.properties.find(prop => prop.name === "webViewId");
+    const scriptProperty = selectedPageEvent.properties.find(prop => prop.name === "script");
+
+    const webViewOptions = this.renderWebViewOptions();
```

Comment:
This seems a bit dirty, maybe you should pass PageLayoutView array to render method (if any) and move handling of empty array before this